invenio_db/shared.py

```diff
@@ -29,9 +33,102 @@
 """Default database metadata object holding associated schema constructs."""
 
 
+class UTCDateTime(TypeDecorator):
+    """Custom UTC datetime type."""
+
+    impl = DateTime
+
+    # todo: should be discussed, but has to be set explicitly to remove warning
+    cache_ok = False
+
+    def process_bind_param(self, value, dialect):
+        """Process value storing into database."""
+        if value is None:
+            return value
+
+        if isinstance(value, str):
+            value = datetime.strptime(value, "%Y-%m-%d %H:%M:%S")
+
+        if not isinstance(value, datetime):
+            msg = f"ERROR: value: {value} is not of type datetime, instead of type: {type(value)}"
+            raise ValueError(msg)
+
+        if value.tzinfo not in (None, timezone.utc):
+            msg = (
+                f"ERror: value: {value} doesn't have a tzinfo of None or timezone.utc."
+            )
+            raise ValueError(msg)
+
+        return value.replace(tzinfo=None)
+
+    def process_result_value(self, value, dialect):
+        """Process value retrieving from database."""
+        if value is None:
+            return None
+
+        if not isinstance(value, datetime):
+            msg = f"ERROR: value: {value} is not of type datetime."
+            raise ValueError(msg)
+
+        if value.tzinfo not in (None, timezone.utc):
+            msg = (
+                f"ERror: value: {value} doesn't have a tzinfo of None or timezone.utc."
+            )
+            raise ValueError(msg)
+
+        return value.replace(tzinfo=None)
```

Comment:
so it should be `tzinfo=timezone.utc`.

even if we are not storing utc (aka "+00:00") explicitly in the database, we assume it is utc and to be explicit what we retrieve from the database should be always explicitly utc (aka "+00:00")